impl/src/main/java/org/ehcache/impl/internal/store/offheap/AbstractOffHeapStore.java

```diff
@@ -133,56 +136,31 @@ public AbstractOffHeapStore(String statisticsTag, Configuration<K, V> config, Ti
-      EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>> map = backingMap();
-      return map == null ? -1L : map.dataVitalMemory();
-    });
-    StatisticsManager.createPassThroughStatistic(this, "mappings", tags, () -> {
-      EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>> map = backingMap();
-      return map == null ? -1L : map.longSize();
-    });
-    StatisticsManager.createPassThroughStatistic(this, "maxMappings", tags, () -> -1L);
-    StatisticsManager.createPassThroughStatistic(this, "vitalMemory", tags, () -> {
-      EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>> map = backingMap();
-      return map == null ? -1L : map.vitalMemory();
-    });
-    StatisticsManager.createPassThroughStatistic(this, "removedSlotCount", tags, () -> {
-      EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>> map = backingMap();
-      return map == null ? -1L : map.removedSlotCount();
-    });
-    StatisticsManager.createPassThroughStatistic(this, "usedSlotCount", tags, () -> {
-      EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>> map = backingMap();
-      return map == null ? -1L : map.usedSlotCount();
-    });
-    StatisticsManager.createPassThroughStatistic(this, "tableCapacity", tags, () -> {
-      EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>> map = backingMap();
-      return map == null ? -1L : map.tableCapacity();
-    });
+    registerStatistic("allocatedMemory", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::allocatedMemory);
+    registerStatistic("occupiedMemory", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::occupiedMemory);
+    registerStatistic("dataAllocatedMemory", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::dataAllocatedMemory);
+    registerStatistic("dataOccupiedMemory", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::dataOccupiedMemory);
+    registerStatistic("dataSize", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::dataSize);
+    registerStatistic("dataVitalMemory", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::dataVitalMemory);
+    registerStatistic("mappings", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::longSize);
+    registerStatistic("vitalMemory", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::vitalMemory);
+    registerStatistic("removedSlotCount", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::removedSlotCount);
+    registerStatistic("usedSlotCount", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::usedSlotCount);
+    registerStatistic("tableCapacity", GAUGE, statisticsTag, EhcacheOffHeapBackingMap::tableCapacity);
 
     this.mapEvictionListener = new BackingMapEvictionListener<>(eventDispatcher, evictionObserver);
   }
 
+  private <T extends Serializable> void registerStatistic(String name, StatisticType type, String tag, Function<EhcacheOffHeapBackingMap<K, OffHeapValueHolder<V>>, T> fn) {
+    StatisticsManager.createPassThroughStatistic(this, name, new HashSet<>(Arrays.asList(name, "tier")), type, () -> {
```

Comment:
Collections.singleton?

Reply:
no because 2 elements but i have replaced by using new static method in StatisticManager.